### PR TITLE
API-8527 take an optional audience arg

### PR DESCRIPTION
**Pull request by crolarlibertyva**
Take an optional audience argument. This is for compatibility across legacy oauth-proxy routes and Keycloak.

Update token fetching to use the /token endpoint defined in the server's metadata.

https://vajira.max.gov/browse/API-8527

### CODE REVIEW

File: auth-cc.js [hunk 1]
```diff
@@ -55,8 +58,13 @@ function getToken(assertion) {
   if (argv["launch"]) {
     body.launch = argv["launch"];
   }
-
-  return axios.post(`${argv["authorization-url"]}/token`, qs.stringify(body), {
```
Comment (dbrown-va) [inline auth-cc.js:59]: This seems to be in addition to the audience change.

However even with the keycloak facing `"api_category": "/fhir1/v1",`, this still works with the old version.

Can the PR description be updated with a call-out for this change as well reasoning behind it?

File: auth-cc.js [hunk 2]
```diff
@@ -72,12 +80,16 @@ function getAudience() {
 
 getAudience()
   .then((response) => {
+    let audience = argv["audience"];
```
Comment (dbrown-va) [inline auth-cc.js:83]: Confirmed this change working with local OAuth proxy to both Okta and Keycloak backed authorization servers.

Before behavior without the ability to specify the audience below.
```
data: {
      error: 'unauthorized_client',
      error_description: "Client authentication with client secret signed JWT failed: FHIR Token audience doesn't match domain. Realm issuer is 'http://docker.for.mac.localhost:9080/fhir-keycloak/realms/fhirdev' but audience from token is '[http://docker.for.mac.localhost:9080/fhir-keycloak/realms/fhirdev/v1/token]'"
    }
```

**Review by dbrown-va — COMMENTED** [review]
Confirmed working with specified audience both native node and dockerized.

Question on the token endpoint change.

File: auth-cc.js (hunk 1 above)
Comment (beaugrantham-va) [reply auth-cc.js:59]: I updated this to always use the metadata for the token URL. This is compatible with both the oauth-proxy routes and Keycloak natively. I'll update the PR comments.

**Review by beaugrantham-va — DISMISSED** [review]

**Review by dbrown-va — APPROVED** [review]

**Pull request merged**